From 7fa1660cf7d7b569e55afc32b1c6272a9a178060 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 15:30:09 -0400
Subject: [block] mtip32xx: Use pci_enable_msix_range() instead of pci_enable_msix()

Message-id: <1402673421-17117-7-git-send-email-snitzer@redhat.com>
Patchwork-id: 83923
O-Subject: [RHEL7.1 PATCH 06/18] mtip32xx: Use pci_enable_msix_range() instead of pci_enable_msix()
Bugzilla: 1105703
RH-Acked-by: David Milburn <dmilburn@redhat.com>

BZ: 1105703

Upstream commit f219ad82f8684334524a089c5dd1c47a9c345a91
Author: Alexander Gordeev <agordeev@redhat.com>
Date:   Wed Feb 19 09:58:16 2014 +0100

    mtip32xx: Use pci_enable_msix_range() instead of pci_enable_msix()

    As result of deprecation of MSI-X/MSI enablement functions
    pci_enable_msix() and pci_enable_msi_block() all drivers
    using these two interfaces need to be updated to use the
    new pci_enable_msi_range() and pci_enable_msix_range()
    interfaces.

    Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
    Cc: Jens Axboe <axboe@kernel.dk>
    Cc: Asai Thambi S P <asamymuthupa@micron.com>
    Cc: linux-pci@vger.kernel.org
    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/block/mtip32xx/mtip32xx.c b/drivers/block/mtip32xx/mtip32xx.c
index 32e4aa3..8573eb6 100644
--- a/drivers/block/mtip32xx/mtip32xx.c
+++ b/drivers/block/mtip32xx/mtip32xx.c
@@ -4619,8 +4619,8 @@ static int mtip_pci_probe(struct pci_dev *pdev,
  INIT_WORK(&dd->work[7].work, mtip_workq_sdbf7);
 
  pci_set_master(pdev);
- rv = pci_enable_msi(pdev);
- if (rv) {
+ rv = pci_enable_msi_range(pdev, 1, 1);
+ if (rv < 0) {
   dev_warn(&pdev->dev,
    "Unable to enable MSI interrupt.\n");
   goto msi_initialize_err;
-- 
1.7.1